src/main/java/org/sagebionetworks/bridge/spring/controllers/HealthDataEx3Controller.java

```diff
@@ -0,0 +1,151 @@
+ * /v1/admin/exporter3
+ */
+@CrossOrigin
+@RestController
+public class HealthDataEx3Controller extends BaseController {
+    private HealthDataEx3Service healthDataEx3Service;
+
+    @Autowired
+    public final void setHealthDataEx3Service(HealthDataEx3Service healthDataEx3Service) {
+        this.healthDataEx3Service = healthDataEx3Service;
+    }
+
+    /** Create or update health data record. Returns the created or updated record. */
+    @PostMapping(path="/v1/admin/exporter3/healthdata")
+    public HealthDataRecordEx3 createOrUpdateRecord() {
+        UserSession session = getAuthenticatedSession(SUPERADMIN);
+        HealthDataRecordEx3 record = parseJson(HealthDataRecordEx3.class);
+        record.setAppId(session.getAppId());
+        HealthDataRecordEx3 savedRecord = healthDataEx3Service.createOrUpdateRecord(record);
+
+        // Write record ID into the metrics, for logging and diagnostics.
+        Metrics metrics = getMetrics();
+        if (metrics != null) {
+            metrics.setRecordId(savedRecord.getId());
+        }
+
+        return savedRecord;
+    }
+
+    /** Deletes all health data records for the given user. */
+    @DeleteMapping(path="/v1/admin/exporter3/participants/{userId}/healthdata")
+    public StatusMessage deleteRecordsForUser(@PathVariable String userId) {
+        UserSession session = getAuthenticatedSession(SUPERADMIN);
+
+        String healthCode = accountService.getHealthCodeForAccount(AccountId.forId(session.getAppId(), userId));
+        if (healthCode == null) {
+            throw new EntityNotFoundException(StudyParticipant.class);
+        }
+
+        healthDataEx3Service.deleteRecordsForHealthCode(healthCode);
+
+        return new StatusMessage("Health data has been deleted for participant");
+    }
+
+    /** Retrieves the record for the given ID. */
+    @GetMapping(path="/v1/admin/exporter3/healthdata/{recordId}")
+    public HealthDataRecordEx3 getRecord(@PathVariable String recordId) {
+        getAuthenticatedSession(SUPERADMIN);
+        HealthDataRecordEx3 record = healthDataEx3Service.getRecord(recordId).orElseThrow(() ->
+                new EntityNotFoundException(HealthDataRecordEx3.class));
+
+        // Write record ID into the metrics, for logging and diagnostics.
+        Metrics metrics = getMetrics();
+        if (metrics != null) {
+            metrics.setRecordId(record.getId());
+        }
+
+        return record;
+    }
+
+    /** Retrieves all records for the given user and time range. */
+    @GetMapping(path="/v1/admin/exporter3/participants/{userId}/healthdata")
+    public ResourceList<HealthDataRecordEx3> getRecordsForUser(@PathVariable String userId,
+            @RequestParam(required = false) String createdOnStart,
+            @RequestParam(required = false) String createdOnEnd) {
+        UserSession session = getAuthenticatedSession(SUPERADMIN);
```

Comment:
This won't be accessible to developers, or worker accounts?

Reply:
That is correct. The purpose of these APIs is specifically to facilitate integration tests. We have no plans to expose these particular APIs outside of this scope.

When in the future, when we build participant and researcher-facing APIs for the Exporter 3 stuff, we'll be creating a new set of APIs, with different names and URLs.